### PR TITLE
Switch clients to ScrollGrid

### PR DESCRIPTION
**Pull request by abarth**
The only remaining client of ScrollableGrid (the old version) is Pesto,
which needs AppBar integration.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: examples/flutter_gallery/lib/demo/grid_list_demo.dart [hunk 1]
```diff
@@ -310,9 +310,6 @@ class GridListDemoState extends State<GridListDemo> {
     });
   }
 
-  // When the ScrollableGrid first appears we want the last row to only be
-  // partially visible, to help the user recognize that the grid is scrollable.
-  // The GridListDemoGridDelegate's tileHeightFactor is used for this.
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/grid_list_demo.dart]: do we no longer do this?

**Comment by Hixie** [comment]
LGTM

File: examples/flutter_gallery/lib/demo/grid_list_demo.dart (hunk 1 above)
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/grid_list_demo.dart]: if we still do it, we should just update the comment to be accurate in the new world

if we don't do it, we should probably make sure we do do it again :-)

Comment (abarth) [inline examples/flutter_gallery/lib/demo/grid_list_demo.dart]: We don't appear to have been doing it.  Do you want me to file a bug about adding that functionality?

Comment (Hixie) [inline examples/flutter_gallery/lib/demo/grid_list_demo.dart]: Yeah, let's file a bug to track it.

**Pull request merged**